ftests/core/src/main/java/org/commonjava/indy/ftest/core/content/AuthoritativeIndexedContentInHostedTest.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright (C) 2013 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.ftest.core.content;
+
+import org.apache.commons.io.IOUtils;
+import org.commonjava.indy.ftest.core.AbstractIndyFunctionalTest;
+import org.commonjava.indy.ftest.core.category.EventDependent;
+import org.commonjava.indy.model.core.HostedRepository;
+import org.commonjava.indy.test.fixture.core.CoreServerFixture;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static org.commonjava.indy.model.core.StoreType.hosted;
+import static org.commonjava.indy.pkg.maven.model.MavenPackageTypeDescriptor.MAVEN_PKG_KEY;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+/**
+ * <b>PREREQUISITES:</b>
+ * <ul>
+ *     <li>org.commonjava.indy.ftest.core.store.HostedAuthIndexWithReadonly must pass</li>
+ * </ul>
+ *
+ * <br />
+ * <b>GIVEN:</b>
+ * <ul>
+ *     <li>Authoritative index of content index enabled</li>
+ *     <li>A hosted repo</li>
+ *     <li>Store one artifact in repo, so this artifact will be cached in index</li>
+ *     <li>Store another artifact in repo through direct file writing, so this artifact will not be cached</li>
+ * </ul>
+ *
+ * <br/>
+ * <b>WHEN:</b>
+ * <ul>
+ *     <li>Change the hosted repo to readonly</li>
+ *     <li>Change the hosted repo back to non-readonly</li>
+ *     <li>Change the hosted repo to readonly again</li>
+ * </ul>
+ *
+ * <br/>
+ * <b>THEN:</b>
+ * <ul>
+ *     <li>When repo is readonly first time, the cached artifact can be fetched, but the direct writing one can not</li>
+ *     <li>When repo is non-readonly, both artifacts are available</li>
+ *     <li>When repo is readonly again, both artifacts are still available</li>
```

Comment:
Because in the second step, when repo non-readonly and the artifacts are fetched, both artifacts have been cached in the content index. So even the repo switched back to readonly again, as they are both in indexing, they are available now.
Maybe I should add one condition for the caching in second step to show why third one happened.